lib/gooddata/lcm/actions/synchronize_user_filters.rb

```diff
@@ -159,46 +166,50 @@ def call(params)
               domain_clients.select! { |c| segments_filter.include?(c.segment_uri) }
             end
 
-            working_client_ids = []
+            working_client_ids = ThreadSafe::Array.new
 
-            filters.group_by { |u| u[multiple_projects_column] }.flat_map do |client_id, new_filters|
+            filters.group_by { |u| u[multiple_projects_column] }.flat_pmap do |client_id, new_filters|
               fail "Client id cannot be empty" if client_id.blank?
-              c = domain.clients(client_id)
-              if params.segments_filter && !segments_filter.include?(c.segment_uri)
-                puts "Client #{client_id} is outside segments_filter #{params.segments_filter}"
+              segment_client = domain_clients.find { |c| c.client_id.to_s == client_id.to_s }
+              if params.segments_filter && !segments_filter.include?(segment_client.segment_uri)
```

Comment:
Identical code found in synchronize_user_filters.rb:174, synchronize_users.rb:256 (mass = 96)